cmd/mimir/help-all.txt.tmpl

```diff
@@ -1145,6 +1145,12 @@ Usage of ./cmd/mimir/mimir:
     	Per-tenant allowed push request burst size. 0 to disable.
   -distributor.request-rate-limit float
     	Per-tenant push request rate limit in requests per second. 0 to disable.
+  -distributor.retry.base duration
+    	[experimental] Base time to wait before retrying a 429/5xx request. (default 3s)
+  -distributor.retry.enabled
+    	[experimental] Enabled controls inclusion of the Retry-After header in the response: true includes it for client retry guidance, false omits it.
+  -distributor.retry.max-allowed-attempts int
+    	[experimental] Sets the upper limit on the number of retry attempts considered for calculation. It caps the calculated attempts without rejecting additional attempts. Used for controlling exponential backoff calculations. (default 5)
```

Comment:
is this per tenant or per client or per distributor? What are some reasonable values? Is 10000 ok? is 2 too small?

If it's too complicated to write in a flag, then we can add a docs page for this

Reply:
ok, now i see it's taking the value of the `retry-attempt` header. we should mention that

Reply:
We can compute max-attempt from max-value internally, right? So for max-value = 150s, we can compute log<base>(max-value) = log3(150) = 4.56. That means max-attempt = 5, but we limit max computed retry time to 150s. WDYT?

Reply:
My math is wrong :) we need `ceil(log2(150/3)) = ceil(5.64) = 6`.

Reply:
I thought about that, but if we do ceil it could end up be more than max, and it might be difficult to document. When user set 150 and then he might think set 160 could eventually improve the situation, but with the math, it will end up change nothing. only by writing this, i feel already dizzy :D

Reply:
I think 160 would work fine... we would still compute max attempts to be `ceil(log2(160/3)) = ceil(5.73) = 6`, but then limit computed retry-after value at 160s instead of 150s.

Reply:
it might be too complex behavior, what do you think the following approach? user want to set to 160, we add validation on max-retry to base * 2 ^ something. And since is not matching the equation, in the error message we propose both the ground and ceil possible values around 160 in this case would be 92 and 184 also we return the equation in error. then user can change the setting accordingly.

Reply:
It would be more intuitive, with a guideline for helping user to set the value with the error message. no guessing involved.

Reply:
> in the error message we propose both the ground and ceil possible values around 160 in this case would be 92 and 184 also we return the equation in error. then user can change the setting accordingly.

as a user, if I want max 160s, why do you force me to set it to 184s?

I don't really see "max value" as very complex or non-intuitive tbh. It's very simple math:

* user sets max to 160s
* we compute max attempts to 6, and use that value internally
* it means that after 5 retries, interval that we use for generating next retry-after will be: [3*2^5 ... 3*2^6], ie. [96, 192].
* when we compute value and it's higher than 160s, we set it to 160s

What could go wrong is if user sets value like 200s. Then we would compute max attempts to be 7, and our last interval would be [3*2^6, 3*2^7], which is [192, 384], but since we limit at 200s, it would be [192s, 200s]. That wouldn't be so great, but it's solveable.

But to keep things simple, we can go with max-attempts for now.

Reply:
> I think 160 would work fine... we would still compute max attempts to be ceil(log2(160/3)) = ceil(5.73) = 6, but then limit computed retry-after value at 160s instead of 150s.

Here do you mean by get the retry attemps by log2(max) then if the result of retry is > 160, set it to 160 for those or set the max to min(base*2^attemp, maxretry)?

Reply:
After discussion, We will keep the retry-max-attemps for the moment and make the document more clear for end user. since it is easier.